addons/mail/static/src/components/message_reaction_group/tests/message_reaction_group_tests.js

```diff
@@ -0,0 +1,175 @@
+    },
+});
+
+QUnit.test('message should have a reaction displayed', async function (assert) {
+    assert.expect(3);
+
+    this.data['mail.channel'].records.push({
+        channel_type: 'channel',
+        id: 20,
+        members: [this.data.currentPartnerId],
+        name: 'test channel',
+        public: 'public',
+    });
+    this.data['mail.message'].records.push({
+        body: 'not empty',
+        id: 10,
+        message_type: 'comment',
+        model: 'mail.channel',
+        res_id: 20,
+    });
+    this.data['mail.message.reaction'].records.push({
+        content: '😊',
+        message_id: 10,
+        partner_id: 7,
+    });
+    const { click, openDiscuss } = await this.start({
+        discuss: {
+            params: {
+                default_active_id: 'mail.channel_20',
+            },
+        },
+    });
+    await openDiscuss();
+    assert.containsOnce(
+        document.body,
+        '.o_MessageReactionGroup',
+        "should have one reaction"
+    );
+    assert.strictEqual(
+        document.querySelector('.o_MessageReactionGroup_content').textContent,
+        "😊",
+        "should have 😊 reaction",
+    );
+    assert.strictEqual(
+        document.querySelector('.o_MessageReactionGroup_count').textContent,
+        "1",
+        "😊 reaction should have only 1 partner having reacted",
+    );
+});
+
+QUnit.test('add reactions to a message', async function (assert) {
+    assert.expect(2);
+
+    this.data['mail.channel'].records.push({
+        channel_type: 'channel',
+        id: 20,
+        members: [this.data.currentPartnerId],
+        name: 'test channel',
+        public: 'public',
+    });
+    this.data['mail.message'].records.push({
+        body: 'not empty',
+        id: 10,
+        message_type: 'comment',
+        model: 'mail.channel',
+        res_id: 20,
+    });
+    this.data['mail.message.reaction'].records.push({
+        content: '😊',
+        message_id: 10,
+        partner_id: 7,
+    });
+    const { click, openDiscuss } = await this.start({
+        discuss: {
+            params: {
+                default_active_id: 'mail.channel_20',
+            },
+        },
+    });
+    await openDiscuss();
+    await afterNextRender(() =>
+        document.querySelector('.o_MessageReactionGroup').click()
+    );
+    assert.strictEqual(
+        document.querySelector('.o_MessageReactionGroup_count').textContent,
+        "2",
+        "😊 reaction should have 2 partners having reacted",
+    );
+    await afterNextRender(() =>
+        document.querySelector('.o_Message').click()
+    );
+    await afterNextRender(() =>
+        document.querySelector('.o_MessageActionList_actionReaction').click()
+    );
+    await afterNextRender(() =>
+        document.querySelector('.o_EmojiList_emoji[data-unicode=\'😃\']').click()
+    );
+    assert.containsN(
+        document.body,
+        '.o_MessageReactionGroup',
+        2,
+        "should have 2 reactions"
+    );
```

Comment:
Isolate this in its own test, which about "adding" another reaction to a reacted message.

Reply:
Done